.githooks/prepare-commit-msg

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+# If the branch name contains the issue number, it will append it to the
+# commit message. Example:
+#
+#   BRANCH NAME               APPEND
+#   feature/PLANET-3125-fix   PLANET-3125
+#   planet-3125               PLANET-3125
+#   some-name                 fail
+
+
+# Customize which branches should be skipped when
+# prepending commit message.
+if [ -z "$BRANCHES_TO_SKIP" ]; then
+  BRANCHES_TO_SKIP=(master develop)
+fi
+
+
+BRANCH_NAME=$(git symbolic-ref --short HEAD)
+PLANET_BRANCH_NAME=$(git symbolic-ref --short HEAD | grep -Po "(PLANET|planet)-[0-9]*" |  tr '[:lower:]' '[:upper:]')
+BRANCH_NAME="${BRANCH_NAME##*/}"
+BRANCH_EXCLUDED=$(printf "%s\n" "${BRANCHES_TO_SKIP[@]}" | grep -c "^$BRANCH_NAME$")
+ISSUE_IN_COMMIT=$(grep -cP "(PLANET|planet)-[0-9]*" $1)
+
+# If planet-xxx is not included in the branch name or the commit message quit.
+# If it is included in the branch name and not in the commit message, prepend it to the commit message.
+if ! [[ -z "$PLANET_BRANCH_NAME" ]] && ! [[ $BRANCH_EXCLUDED -eq 1 ]] && ! [[ $ISSUE_IN_COMMIT -ge 1 ]]; then
+  sed -i.bak -e "1s/^/$PLANET_BRANCH_NAME /" $1;
+elif [[ -z "$PLANET_BRANCH_NAME" ]] && [[ $ISSUE_IN_COMMIT -eq 0 ]]; then
+  echo "Error! Issue number is not part of the branch name or the commit message.";
+  echo "Commit message should start with PLANET-xxx or branch name should contain planet-xxx or PLANET-xxx.";
+  exit 1;
+fi
```

Comment:
There's a lot of if conditions here which could probably be simplified, for example instead of an elif, test for the conditions in line 29 first and then use a separate if statement afterwards.  This reduces [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) and increases maintanability.